### PR TITLE
task/256

### LINKED ISSUE #256: Test PictogramController
We would like to have the PictogramController tested

Possible Suggested Solution

Create a PictogramControllerTest.cs file
Write unit tests, by mocking the repository and the connection to the controller ( see repository pattern https://tinyurl.com/2p8sadnr)

### PR DESCRIPTION
**Pull request by manfredhod**
# Description

Added the necessary tests for the PictrogramController after it had been refactored. 

Fixes #256 

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

The refactor works, and all the added unit tests pass. 

**Development Configuration**

Dotnet version: 6

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [x] I have Acceptance Tested this on an Android device

### CODE REVIEW

**Review by Jaffenheimer — APPROVED** [review]
Code looks good to me, all pictogramcontroller unit-tests passing when running locally, approving this.

**Review by andreaslborg — CHANGES_REQUESTED** [review]
Please look into the integration test. This is yours:
`Failed!  - Failed:    42, Passed:   122, Skipped:     0, Total:   164`
Compared to main:
`Failed!  - Failed:     4, Passed:   133, Skipped:     0, Total:   137`

Also, see this PR: https://github.com/aau-giraf/web-api/pull/312
Their integration tests pass: 
`Passed!  - Failed:     0, Passed:   137, Skipped:     0, Total:   137`

File: GirafRest.Test/appsettings.Development.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿{
     "_Comment": "ONLY FOR TESTING PURPOSE",
     "ConnectionStrings": {
-        "DefaultConnection": "server=localhost;port=3306;userid=user;password=password;database=giraf;Allow User Variables=True"
```
Comment (andreaslborg) [inline GirafRest.Test/appsettings.Development.json]: Also, is it intentional to change user -> root here?

**Review by andreaslborg — CHANGES_REQUESTED** [review]

**Comment by Jaffenheimer** [comment]
Skulle nok have kørt alle tests i stedet for kun dem de havde lagt ind haha

File: GirafRest.Test/appsettings.Development.json (hunk 1 above)
Comment (manfredhod) [reply GirafRest.Test/appsettings.Development.json]: Fixed.

**Comment by manfredhod** [comment]
> Please look into the integration test. This is yours: `Failed! - Failed: 42, Passed: 122, Skipped: 0, Total: 164` Compared to main: `Failed! - Failed: 4, Passed: 133, Skipped: 0, Total: 137`
> 
> Also, see this PR: #312 Their integration tests pass: `Passed! - Failed: 0, Passed: 137, Skipped: 0, Total: 137`

Should be updated now.

**Review by andreaslborg — APPROVED** [review]

**Pull request merged**